test/test_web_server.rb

```diff
@@ -79,10 +59,30 @@ def test_header_is_too_long
     end
   end
 
+  # TODO: Why does this test take exactly 20 seconds?
   def test_file_streamed_request
     body = "a" * (Puma::Const::MAX_BODY * 2)
     long = "GET /test HTTP/1.1\r\nContent-length: #{body.length}\r\n\r\n" + body
-    do_test(long, (Puma::Const::CHUNK_SIZE * 2) - 400)
+    socket = do_test(long, (Puma::Const::CHUNK_SIZE * 2) - 400)
+    assert_match "hello", socket.read
   end
 
+  private
+
+  def do_test(string, chunk, close_after=nil)
+    # Do not use instance variables here, because it needs to be thread safe
```

Comment:
Not sure this is true.

Reply:
I think a new instance is used in every thread?  If so, instance variables should be ok.  JFYI, I'm running `test_puma_server_ssl.rb` right now locally, and it passed running parallel...